src/bin/nydus-image/core/context.rs

```diff
@@ -37,25 +37,46 @@ use super::prefetch::{Prefetch, PrefetchPolicy};
 // TODO: select BufWriter capacity by performance testing.
 pub const BUF_WRITER_CAPACITY: usize = 2 << 17;
 
-#[derive(Clone, Copy, Debug, PartialEq)]
-pub enum SourceType {
-    Directory,
-    StargzIndex,
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub enum ConversionType {
+    DirectoryToRafs,
+    DirectoryToStargz,
```

Comment:
Maybe we need to call it `EStargz`, it's a different format from stargz (it's not a blocker for this PR).

Reply:
Will fix in later patch.